apps/passport-client/src/appHooks.ts

```diff
@@ -168,3 +169,32 @@ export function useRequirePassword() {
     });
   }
 }
+
+// Hook that enables keystrokes to properly listen to laser scanning inputs from supported devices
+export function useLaserScannerKeystrokeInput() {
+  const [typedText, setTypedText] = useState("");
+
+  useEffect(() => {
+    function handleKeyDown(event: KeyboardEvent) {
+      if (event.key === "Enter") {
+        // Check URL regex and navigate to the last match, if it exists
+        const url = getLastValidURL(typedText);
+        if (url) {
+          window.location.href = url;
+        }
+      }
+      // Ignore characters that could not be in a valid URL
+      if (/^[a-zA-Z0-9\-._~!$&'()*+,;=:@%#/]$/.test(event.key)) {
```

Comment:
are you sure `:` can't be in a valid url? i'm suspicious of this regex.

Reply:
you need it for the `:` in `https:` or `http:`